crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -343,6 +344,15 @@ fn check_one_object(
                 }
             };
         }
+        InputObjectKind::SharedMoveObject {
```

Comment:
Should we do anything similar for the system object? Is it only mutated in system transactions?

Reply:
No, sadly not -- (at least not currently) -- delegation requests will touch it, IIRC.